.pre-commit-config.yaml

```diff
@@ -1,60 +1,42 @@
 repos:
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v3.15.0
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.6.7
     hooks:
-      - id: pyupgrade
-        args: [--py39-plus]
+      - id: ruff
+        args: [--exit-non-zero-on-fix]
 
   - repo: https://github.com/psf/black-pre-commit-mirror
-    rev: 23.12.1
+    rev: 24.8.0
     hooks:
       - id: black
 
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.13.2
-    hooks:
-      - id: isort
-        args: [--add-import=from __future__ import annotations]
```

Comment:
You can also do this with Ruff (if you like) via the `tool.ruff.lint.isort.required-imports` setting: https://docs.astral.sh/ruff/settings/#lint_isort_required-imports

Reply:
Re-added!